### DISCUSSION TITLE
Would Mockoon team consider to support HTTP/HTTPS request(like curl) as helper functions ?

### DISCUSSION THREAD

**Discussion by elianre** [mockoon/mockoon#1286] [Ideas] [answered]
Dear Guillaume,

I really like the design of Mockoon and would like to use it for my daily work. It providers comprehensive functions to deal with the requests and generate response to satisfy different kind of requirements. However, for my daily work, I need more than that. I need the mock service can send out some HTTP/HTTPS request like doing a curl after it receives the request, then generate the response according to the curl response too. So would the team consider to support this kind of requirement in the future?

Thanks,
Rex

**Comment by 255kb** [comment, marked as answer]
Hi, while I understand the use case, it's not something that is planned now. This would require to introduce a new data fetching solution different from the current callbacks feature, as Mockoon would have to wait for the answer to be able to reuse it in the initial response. I think this is also going a bit too far from the usual API mocking features.

Thank you.

PS: similar [request](https://github.com/mockoon/mockoon/discussions/1287).
